docs/auth/keycloak/keycloak-upstream-IAMs-conf.md

```diff
@@ -0,0 +1,97 @@
+![Github Keycloak](diagrams/github-keycloak.png)
+
+## Setup Google as an Upstream Identity Provider
+If you don't have a Google account, follow the instrctions below to create one. 
+> [!NOTE] For simplicity we will be creating an OAUTH app under a personal google account.
+- Go to Google Cloud Platform console: https://console.cloud.google.com
+![Google Cloud Console](diagrams/google-cloud-console.png)
+- Go to `APIs & Services` -> `OAuth consent screen` tab on left hand side and select `External` user type consent screen to create a new consent screen. 
+- After creating the consent screen, go to `Credentials` tab on left hand side, click on `CREATE CREDENTIALS` and select `OAuth Client ID`. For `Application type` select `Web application`. Give it a client name, for `Authorized redirect URIs` use the Redirect URI in your keycloak console. 
+- Once you click create note your client id and client secret. 
+![Google Cloud Credentials](diagrams/google-cloud-credentials.png)
+
+### Keycloak Google setup 
+Go back to keycloak console and select `Google` as an identity provider. 
+> [!IMPORTANT] Make sure you are in the `tornjak` realm or te specific realm Tornjak app is registered on 
+- The `Redirect URI` is automatically set by keycloak, this is the uri you should enter in the `Authorized redirect URIs` while creating your google OAuth app above. 
+- Copy ad paste the `Client ID` and `Client Secret` you generated in the google OAuth app from above in the respective fields. 
+- And click `Add`
+- Now when you try signing in to the Tornjak application, you should see the keycloak login page and google as an optional upstream identity provider to sign in to. If you sleect google, keycloak will redirect you to sign in with google login and tornjak will be authenticated using google OAuth. 
+> [!IMPORTANT] Make sure you assign appropraite roles within keycloak for your user, or the roles are mapped correctly between google and keycloak. 
+![Google Keycloak](diagrams/google-keycloak.png)
+
+## Setup Openshift as an Upstream Identity Provider
+If you don't have an Openshift account, follow the instrctions below to create one. 
+
+****DOCUMENTATION FOR OPENSHIFT TO BE ADDED!!!****
+
+- As you add an upstream IAM, you will see an option to sign in with the IAM in the keycloak login page as shown on the picture below. 
+![Keycloak Sign In](diagrams/keycloak-sign-in-page.png)
+
+> [!TIP] To make an upstream identity provider default: in another words for keycloak to directly redirect to the choosen identity provider, instead of showing the default keycloak login page follow the following steps below:
+- Click on `Authentication` on the left handside menu
+- Choose the `browser` flow
+![Keycloak Browser Flow](diagrams/browser-flow.png)
+- Click on the gear icon on the `Identity Provider Redirector` and set to the alias of the identity provider you want keycloak to redirect to. 
```

Comment:
Make this a separate section perhaps to separate this note from the Openshift for IAM section